spec/unit/blockscore/question_set_spec.rb

```diff
@@ -1,62 +1,210 @@
     end
 
-    describe '#all' do
-      it 'requests' do
-        person.question_sets.all
-        assert_requested(api_stub, times: 2)
+
+    describe "#new" do
+
+      subject { person.question_sets }
+
+      it "should delegate to QuestionSet" do
+        expect(QuestionSet).to receive(:new).with(subject.default_params).and_call_original
+        result = person.question_sets.new
+        expect(result).to be_an_instance_of(QuestionSet)
       end
 
-      it ':count' do
-        response = person.question_sets.all(count: 2)
-        expect(response.count).to eq(2)
-        assert_requested(api_stub, times: 2)
+      it "should save person if not save" do
+        person = Person.new
+        qs = person.question_sets.new
+        expect(person).to receive(:save).and_call_original
+        qs.save
       end
 
-      it ':offset' do
-        response = person.question_sets.all(count: 2, offset: 2)
-        expect(response.count).to eq(2)
-        assert_requested(api_stub, times: 2)
+      it "should have person#id" do
+        qs = person.question_sets.new
+        qs.person_id = "some_id"
+        qs.save
+        expect(qs.person_id).to eq(person.id)
+        expect(qs.id).not_to be(nil)
       end
+
+      it "should add to question_sets" do
+        count = subject.size
+        qs = subject.new
+        expect(subject.size).to eq(count + 1)
+        expect(subject).to include(qs)
+      end
+
     end
 
-    describe '#score' do
-      let(:answers) do
-        [
-          { question_id: 1, answer_id: 1 },
-          { question_id: 2, answer_id: 1 },
-          { question_id: 3, answer_id: 1 },
-          { question_id: 4, answer_id: 1 },
-          { question_id: 5, answer_id: 1 }
-        ]
+    describe "#all" do
+      subject { person.question_sets.all }
+      it "should return self" do
+        is_expected.to be(person.question_sets)
+      end
+    end
+
+    describe "#create" do
+
+      let (:person) { create(:person) }
+      let (:data) { person.attributes[:question_sets] }
+      let (:qs) { QuestionSet.create({person_id: person.id}) }
+      subject { person.question_sets.create }
+
+      it "should pass the Person#id" do
+        expect(QuestionSet).to receive(:create).with({person_id: person.id}).and_call_original
+        subject
+      end
+
+      it "should update Person#attributes" do
+        expect(data).to include(subject.id)
+      end
+
+      it "should update question_sets collection" do
+        expect(person.question_sets).to include(subject)
+      end
+
+      it "should return question set" do
+        is_expected.to be_an_instance_of(QuestionSet)
       end
 
-      it 'score call does request' do
-        qs = person.question_sets.create
-        qs.score(answers)
-        assert_requested(api_stub, times: 3)
+      it "should error if parent not created" do
+        person = Person.new
+        expect { person.question_sets.create }.to raise_error(Error, "Create parent first")
       end
```

Comment:
Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.